pkg/engine/shield/add_test.go

```diff
@@ -0,0 +1,362 @@
+package shield
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/simimpact/srsim/tests/mock"
+)
+
+// Unit Tests for AddShield()
+func TestShieldHealthByPositiveValues(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	engine := mock.NewMockEngineWithEvents(mockCtrl)
```

Comment:
`MockEngine` is not necessary here. Can just create your own event system via `events := &event.System{}`